src/NuGet.Core/NuGet.CommandLine.XPlat/Program.cs

```diff
@@ -69,6 +73,51 @@ public static int MainInternal(string[] args, CommandOutputLogger log)
 
             NuGet.Common.Migrations.MigrationRunner.Run();
 
+            if (args.Count() >= 2 && args[0] == "package" && args[1] == "search")
+            {
+                // We are executing command `dotnet package search`
+                Func<ILoggerWithColor> getHidePrefixLogger = () =>
+                {
+                    log.HidePrefixForInfoAndMinimal = true;
+                    return log;
+                };
+
+                CliCommand rootCommand = new CliCommand("package");
+                PackageSearchCommand.Register(rootCommand, getHidePrefixLogger);
+
+                args[0] = null;
+                args = args
+                    .Where(e => e != null)
+                    .ToArray();
+
+                foreach (CliCommand command in rootCommand.Subcommands)
```

Comment:
Does `await rootCommand.InvokeAsync(args)` (or something like that) make this loop redundant?

Reply:
fixed